proximal/examples/test_deconv_sv_psf.py

```diff
@@ -61,26 +61,26 @@ def weighted_planewise_conv(G, W, im_shape, n_psf: int):
     weight = W[:n_psf].reshape((n_psf, *im.shape))
 
     def forward(im_src, im_sim):
-        assert np.all(im_src.shape == im_sim.shape[1:])
-        assert im_sim.shape[0] == n_psf
+        assert np.all(im_src.shape == im_sim.shape)
 
+        im_sim[:] = 0
         for i in range(n_psf):
-            im_sim[i, :] = fftconvolve(psf_mode[i],
-                                       im_src * weight[i],
-                                       mode='same')
+            im_sim[:] += fftconvolve(psf_mode[i],
+                                     im_src * weight[i],
+                                     mode='same')
 
     def adjoint(im_src, im_sim):
-        assert im_src.shape[0] == n_psf
-        assert np.all(im_src.shape[1:] == im_sim.shape)
+        assert np.all(im_src.shape == im_sim.shape)
 
         im_sim[:] = 0
         for i in range(n_psf):
             im_sim[:] += fftconvolve(
-                im_src[i], np.rot90(psf_mode[i], k=2), mode='same') * weight[i]
+                np.rot90(psf_mode[i], k=2), im_src, mode='same') * weight[i]
```

Comment:
> ... but its giving me a ValueError about array having inhomogeneous shape, which is being raised inside proximal/algorithms/problem.py during a list comprehension on prox_fns.

No I didn't encounter the `ValueError`. I am using numpy == 1.17.4. I like this error, coz it flushes out my earlier problem modelling mistake. I will upgrade my own Numpy later.